hnn_core/network.py

```diff
@@ -1253,23 +1253,31 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
-    def clear_connectivity(self):
-        """Remove all connections defined in Network.connectivity
-        """
+    def _clear_connectivity(self, src_types=None):
```

Comment:
Actually I wonder if we should just combine this with the `pick_connection` API?

Reply:
Possibly, but for now I made it private so I don't need to answer that question ;-)

Reply:
You have two options: expose all the arguments that are in `pick_connection`. But this needs to be done in a consistent way. The other is to allow passing keyword arguments but then the documentation is not so nice because user will have to go to `pick_connection` function to check the documentation of `clear_connectivity`. But for now it's private ...